src/nvim/message.c

```diff
@@ -137,6 +139,94 @@ int msg_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
   return msg_attr_keep((char_u *)s, attr, false);
 }
 
+int msg_echo_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
+{
+  return msg_echo_attr_keep(s, attr, false);
+}
+
+int
+msg_echo_attr_keep(
+    const char *s,
+    const int attr,
+    int keep
+)
+  FUNC_ATTR_NONNULL_ARG(1)
```

Comment:
FUNC_ATTR_NONNULL_ALL may be used when all *pointer* arguments are non-NULL. Non-pointer ones do not matter.